doc/src/overrides.md

```diff
@@ -111,7 +111,8 @@ that uses the new TOML encoding in the `rust-toolchain` file. You need to upgrad
 `rustup` to 1.23.0+.
 
 The `rust-toolchain.toml`/`rust-toolchain` files are suitable to check in to
-source control.
+source control. If that's done, `Cargo.lock` should probably be tracked too to
```

Comment:
Thanks for your contribution.

Does this conflict with https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries?

Reply:
Does it?

Arguments for/against `Cargo.lock` for libraries apply also to `rust-toolchain`, no?

> The purpose of a Cargo.lock lockfile is to describe the state of the world at the time of a successful build. Cargo uses the lockfile to provide deterministic builds on different times and different systems, by ensuring that the exact same dependencies and versions are used as when the Cargo.lock file was originally generated.

And the toolchain used is a part of *the state of the world*. So this, in my mind, reinforces the tie between both files.

> Users dependent on the library will not inspect the library’s Cargo.lock (even if it exists).

Same can be said about `rust-toolchain`, no?

> In other words, libraries specify SemVer requirements for their dependencies but cannot see the full picture. Only end products like binaries have a full picture to decide what versions of dependencies should be used.

.. and what toolchain version should be used?

----

So, one can argue that, ideally, libraries shouldn't track either file. But, and irregardless of crate type, tracking one of them files signals a desire to be able to re-create *the state of the world* in the future easily. And to truly be able to re-create *the state of the world*, you need both files!

Reply:
What I mean is that we should not recommend in this book about the way `Cargo.lock` is handled. It should follow the rules of the documentation I posted.

Reply:
I think its true that for users of a library crate, the toolchain files interactions with dependencies selected by cargo are irrelevant. But I also think its true that for the given scenario - a version fixed toolchain file, then developer of a library crate, doing a clean checkout will often break because dependencies have moved on but rust/cargo are pinned at an ancient version. So documenting that friction and giving people pragmatic advice - 'if you use a concrete version locked toolchain file then also versioning Cargo lock' seems wise.